include/vga.h

```diff
@@ -755,6 +755,84 @@ union CrtcModeControlRegister {
 	bit_view<7, 1> is_sync_enabled;
 };
 
+// Attribute Controller Registers
+// ==============================
+//
+// The Attribute Registers consist of:
+// - the sixteen Palette Registers,
+// - the Mode Control Register,
+// - the Overscan Color Register,
+// - the Color Plane Enable Register,
+// - and the Horizontal Pixel Panning Register.
+//
+// The VGA also includes the Color Select Register.
+
+// Attribute Address Register
+// --------------------------
+// (write port: 3C0h on EGA & VGA, read port: 3C1h on VGA only)
+//
+// Selects the Attribute Controller Registers that will be selected during a
+// write operation for EGA/VGA, or a read operation for VGA.
+//
+// The attribute controller has only one port dedicated to it at 3C0h. An
+// internal flip-flop is used to multiplex this port to load either this Address
+// Attribute Register or one of the Attribute Registers:
+//
+// - When the flip-flop is in the clear state, it causes port writes
+//   3COh to be directed to this Address Attribute Register.
+//
+// - When the flip-flop is in the set state, data written to this port is
+//   directed to whichever Attribute Register index is loaded into the
+//   attribute_address field of this register.
+//
+union AttributeAddressRegister {
+	uint8_t data = 0;
+
+	// Points to one of the Attribute Address Registers:
+	// 00h-0Fh - Palette Registers 0-15
+	// 10h     - Mode Control Register
+	// 11h     - Overscan Color Register
+	// 12h     - Color Plane Enable Register
+	// 13h     - Horizontal Pixel Panning Register
+	// 14h     - Color Plane Enable (VGA only)
+	bit_view<0, 5> attribute_address;
+
+	// Determines whether the palette dual-ported RAM should be accessed by
+	// the host or by the EGA display memory.
+	//
+	// 0: Allows the host to access the palette RAM. Disables the display
+	// memory from gaining access to the palette.
+	//
+	// 1: Allows the display memory to access the palette RAM. Disables the
+	// host from gaining access to the palette.
+	bit_view<5, 1> palette_address_source;
+};
+
+// Palette Registers (index 00h-0Fh)
+union PaletteRegister {
+	uint8_t data = 0;
+
+	// On EGA, the values describe the colours directly. sr, sg, and sb stand
+	// for the "secondary" RGB values, forming basically a 2-bit RGB colour
+	// codes:
+	//
+	// sr,sg,sb   r,g,b    saturation
+	//     0        0          0%
+	//     0        1         33%
+	//     1        0         66%
+	//     1        1        100%
+	bit_view<0, 1> b;
+	bit_view<1, 1> g;
+	bit_view<2, 1> r;
+	bit_view<3, 1> sb;
+	bit_view<4, 1> sg;
+	bit_view<5, 1> sr;
```

Comment:
hey.. another failing of C (and C++, damnit! :face_exhaling:) - unreliable bitfields. 

At least we've got our own. :laughing: